### DISCUSSION TITLE
Ties show up twice in Snake History

### DISCUSSION THREAD

**Discussion by coreyja** [BattlesnakeOfficial/feedback#54] [Bug Reports]
When on the page for you snake. For instance: [https://play.battlesnake.com/u/coreyja/devious-devin/](https://play.battlesnake.com/u/coreyja/devious-devin/)

If there was a tie in an Arena (only seen with Global Duels) the game will display twice in History. Both links have the same `id` and go to the same page

![image](https://user-images.githubusercontent.com/964771/119281703-a50e4900-bc04-11eb-9ccf-48cb800acc59.png)

**Comment by coreyja** [comment]
I've since seen this happen in both Arena and Royale games. I also don't think its _only_ when I tie. It _could_ be if any snake ties, but that could also just have been a red herring originally

**Reply by chris-bsnake** [reply]
That is strange. I'll added it to the bug shortlist for this week for investigation

**Reply by chris-bsnake** [reply]
We found the source of the issue. There is a race condition between a primary and a backup systems that checks game status in the engine and verifies completed games, and sometimes they both can log a game result record at the same time.  I've done some tweaks that should reduce the likelihood of this happening.  Strangely enough it has nothing to do with tie games, but it does happen more often for arena games because so many of them are generated each day.

The tweak should go live with release v1.5.45

**Reply by coreyja** [reply]
Thanks for the explanation!
Always fun to learn how bug like this pop up!

**Comment by bvanvugt** [comment]
This was resolved in release 1.5.45! Thanks for reporting :)
